rcljava/include/org_ros2_rcljava_contexts_ContextImpl.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
```

Comment:
I don't know how I feel about this. I'd rather change all the copyright notices (the ones that are right now in master and any future notices) to something generic like "ros2-java developers", instead of adding OSRF. I prefer to keep ros2-java separate from OSRF, as it may lead people to think that it's an OSRF project.

Reply:
I think if an individual or organization contributes a substantial change to a source file they should be able to add their name to the copyright notice. It can act as an incentive for contributors. For example, this is a common practice we follow for packages in the [ros2](https://github.com/ros2) org. 

Alternatively, if there's preference to add a generic notice to every file, I think it'd be nice to also add a "CONTRIBUTORS" file listing major contributors. Contributors are tracked with version control, but if the system changes or project migrates far in the future at least there's a file record.

I don't know the legal implications of either option. But, my personal preference is to list each copyright owner in every file since it looks less ambiguous.

Reply:
> I think if an individual or organization contributes a substantial change to a source file they should be able to add their name to the copyright notice. It can act as an incentive for contributors.

But that's not what copyright noticed are for. They are not for acknowledging contributors, but to state the terms of how the software is distributed.

> For example, this is a common practice we follow for packages in the ros2 org.

That's interesting. What I remember from my days at OSRF is that it respected the copyright notices of the ROS 1 packages, despite having taken over of stewardship of it.

> Alternatively, if there's preference to add a generic notice to every file, I think it'd be nice to also add a "CONTRIBUTORS" file listing major contributors. Of course, individual contributions are tracked with version control, but if the version control system changes or project migrates far in the future at least there's a file record.

I like that more, and there's less arguing about what constitutes a substantial change.

Reply:
> What I remember from my days at OSRF is that it respected the copyright notices of the ROS 1 packages, despite having taken over of stewardship of it

For ports of ROS 1 packages, we've kept the same license, but often added multiple copyright notices where substantial changes were made (e.g. new features). For example, in RViz: https://github.com/ros2/rviz/blob/5595a69398df4d93aea711d408772ea34c53c230/rviz_common/include/rviz_common/factory/factory.hpp#L2-L4

---

See https://github.com/ros2-java/ros2_java/pull/124 for a proposed change to the copyright notices.

Reply:
> But that's not what copyright noticed are for. They are not for acknowledging contributors, but to state the terms of how the software is distributed.

For what its worth, the style of adding the copyright owner at the top of the license is a very standard practice.  For instance, https://opensource.org/licenses/Apache-2.0 explains on adding the license, a copyright owner should be added.  In most projects I've seen (including ROS 2), the original author of the file is the copyright owner, but I don't have documentation for that, just anecdotal evidence.

The other way I've seen to do it is to make the copyright owner "The FOO Authors", and then list the contributors separately as @jacobperron mentioned (this is how Cartographer [does it](https://github.com/cartographer-project/cartographer/blob/master/cartographer/io/fake_file_writer_test.cc)).

Reply:
After discussing "offline", we've decided on a policy that new files may contain a different copyright notice while existing files should not have their copyright notice changed or added to.

Reply:
I found one file that was in contradiction to the policy and fixed it: d270e67